### ISSUE TITLE
The "--output gruff" option has a label name incorrect if a context is specified.

### ISSUE THREAD

**Issue by naitoh** [benchmark-driver/benchmark-driver#39]
The "--output gruff" option has a label name incorrect if a context is specified.

# No Problem case 1 : output simple : ruby versions

```
$ cat test_ruby.yaml
loop_count: 2000000
prelude: |
  class Array
    alias_method :blank?, :empty?
  end
  array = []
benchmark:
  empty: array.empty?
  blank: array.blank?
```

```
$  benchmark-driver test_ruby.yaml --output simple  --rbenv '2.4.1;2.5.1'
Iteration per second (i/s):
            2.4.1       2.5.1  
empty     96.502M     78.542M  
blank     42.993M     50.283M  
```

It is a correct.

# No Problem case 2  :  output simple : gem versions

```
$ cat test_gem.yaml
contexts:
  - gems: { red-datasets : 0.0.3 }
    require: false
    prelude: require 'datasets'
  - gems: { red-datasets : 0.0.4 }
    require: false
    prelude: require 'datasets'

loop_count: 2000000
prelude: |
  class Array
    alias_method :blank?, :empty?
  end
  array = []
benchmark:
  empty: array.empty?
  blank: array.blank?
```

```
$  benchmark-driver test_gem.yaml --output simple
Iteration per second (i/s):
       red-datasets 0.0.3  red-datasets 0.0.4  
empty     88.304M    124.922M  
blank     35.911M     39.038M  
```

It is a correct.

# No Problem case 3 : output gruff : ruby versions

```
$  benchmark-driver test_ruby.yaml --output gruff  --rbenv '2.4.1;2.5.1'
* empty...
  * 2.4.1...
  * 2.5.1...
* blank...
  * 2.4.1...
  * 2.5.1...
rendering graph...: graph.png
```
![graph_ruby](https://user-images.githubusercontent.com/706220/42510636-85b6fed0-848a-11e8-86fa-4bebcb08d89c.png)

It is a correct.

# Problem case 1 : output gruff : gem versions

```
$  benchmark-driver test_gem.yaml --output gruff  
* empty...
  * red-datasets 0.0.3...
  * red-datasets 0.0.4...
* blank...
  * red-datasets 0.0.3...
  * red-datasets 0.0.4...
rendering graph...: graph.png
```

![graph_gem](https://user-images.githubusercontent.com/706220/42510702-b9361dcc-848a-11e8-96b8-9c3b3b9e3110.png)

Label title is wrong.
I expect gem version label.

# Problem case 2 : specify multiple versions.

I specified multiple ruby version numbers and multiple gem version numbers at the same time.
 Is this case supported?

```
$  benchmark-driver test_gem.yaml --output simple  --rbenv '2.4.1;2.5.1'
Iteration per second (i/s):
       red-datasets 0.0.3 (2.4.1)  red-datasets 0.0.3 (2.5.1)  red-datasets 0.0.4 (2.4.1)  red-datasets 0.0.4 (2.5.1)  
empty    216.333M     71.521M  /Users/naitoh/.rbenv/versions/2.4.1/lib/ruby/2.4.0/rubygems/dependency.rb:310:in `to_specs': Could not find 'red-datasets' (= 0.0.4) - did find: [red-datasets-0.0.3] (Gem::MissingSpecVersionError)
Checked in 'GEM_PATH=/Users/naitoh/.gem/ruby/2.4.0:/Users/naitoh/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0', execute `gem env` for more information
	from /Users/naitoh/.rbenv/versions/2.4.1/lib/ruby/2.4.0/rubygems/dependency.rb:320:in `to_spec'
	from /Users/naitoh/.rbenv/versions/2.4.1/lib/ruby/2.4.0/rubygems/core_ext/kernel_gem.rb:65:in `gem'
	from /var/folders/h4/xw9c7n5957x84431pqbrx6xh0000gn/T/benchmark_driver-20180710-75330-125zyz0.rb:1:in `<main>'
     ERROR    317.460M  
blank     52.207M     40.732M  /Users/naitoh/.rbenv/versions/2.4.1/lib/ruby/2.4.0/rubygems/dependency.rb:310:in `to_specs': Could not find 'red-datasets' (= 0.0.4) - did find: [red-datasets-0.0.3] (Gem::MissingSpecVersionError)
Checked in 'GEM_PATH=/Users/naitoh/.gem/ruby/2.4.0:/Users/naitoh/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0', execute `gem env` for more information
	from /Users/naitoh/.rbenv/versions/2.4.1/lib/ruby/2.4.0/rubygems/dependency.rb:320:in `to_spec'
	from /Users/naitoh/.rbenv/versions/2.4.1/lib/ruby/2.4.0/rubygems/core_ext/kernel_gem.rb:65:in `gem'
	from /var/folders/h4/xw9c7n5957x84431pqbrx6xh0000gn/T/benchmark_driver-20180710-75330-wxenmd.rb:1:in `<main>'
     ERROR     40.393M  
```

```
$  benchmark-driver test_gem.yaml --output gruff  --rbenv '2.4.1;2.5.1'
* empty...
  * red-datasets 0.0.3 (2.4.1)...
  * red-datasets 0.0.3 (2.5.1)...
/Users/naitoh/.rbenv/versions/2.4.1/lib/ruby/2.4.0/rubygems/dependency.rb:310:in `to_specs': Could not find 'red-datasets' (= 0.0.4) - did find: [red-datasets-0.0.3] (Gem::MissingSpecVersionError)
Checked in 'GEM_PATH=/Users/naitoh/.gem/ruby/2.4.0:/Users/naitoh/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0', execute `gem env` for more information
	from /Users/naitoh/.rbenv/versions/2.4.1/lib/ruby/2.4.0/rubygems/dependency.rb:320:in `to_spec'
	from /Users/naitoh/.rbenv/versions/2.4.1/lib/ruby/2.4.0/rubygems/core_ext/kernel_gem.rb:65:in `gem'
	from /var/folders/h4/xw9c7n5957x84431pqbrx6xh0000gn/T/benchmark_driver-20180710-75609-1nu4ou6.rb:1:in `<main>'
  * red-datasets 0.0.4 (2.4.1)...
  * red-datasets 0.0.4 (2.5.1)...
* blank...
  * red-datasets 0.0.3 (2.4.1)...
  * red-datasets 0.0.3 (2.5.1)...
/Users/naitoh/.rbenv/versions/2.4.1/lib/ruby/2.4.0/rubygems/dependency.rb:310:in `to_specs': Could not find 'red-datasets' (= 0.0.4) - did find: [red-datasets-0.0.3] (Gem::MissingSpecVersionError)
Checked in 'GEM_PATH=/Users/naitoh/.gem/ruby/2.4.0:/Users/naitoh/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0', execute `gem env` for more information
	from /Users/naitoh/.rbenv/versions/2.4.1/lib/ruby/2.4.0/rubygems/dependency.rb:320:in `to_spec'
	from /Users/naitoh/.rbenv/versions/2.4.1/lib/ruby/2.4.0/rubygems/core_ext/kernel_gem.rb:65:in `gem'
	from /var/folders/h4/xw9c7n5957x84431pqbrx6xh0000gn/T/benchmark_driver-20180710-75609-4rsc0b.rb:1:in `<main>'
  * red-datasets 0.0.4 (2.4.1)...
  * red-datasets 0.0.4 (2.5.1)...
rendering graph...: graph.png
```

![graph_ruby_gems](https://user-images.githubusercontent.com/706220/42510797-03ab57a0-848b-11e8-82c2-d109664468e3.png)

Label title is wrong.

## Environment

* macOS Sierra
* ruby 2.5.1p57 (2018-03-29 revision 63029) [x86_64-darwin16]
* benchmark_driver (0.14.2)
* benchmark_driver-output-gruff (0.3.0)

Best regards.

**Comment by k0kubun** [comment]
Thanks to use these features and report them!

> Problem case 1 : output gruff : gem versions

Please report this to https://github.com/benchmark-driver/benchmark_driver-output-gruff/issues/new next time. 

This was a bug of the plugin. Fixed in v0.3.1. Please try it.

> Problem case 2 : specify multiple versions.

This is not a bug in both benchmark_driver and benchmark_driver-output-gruff. You just need install them in each Ruby versions. 

For now benchmark_driver does not support to install required gems automatically for your Rubies. I recommend to install rbenv-each and do:

```bash
rbenv each gem install red-datasets -v 0.0.3
rbenv each gem install red-datasets -v 0.0.4
```

**Comment by naitoh** [comment]
> Please report this to https://github.com/benchmark-driver/benchmark_driver-output-gruff/issues/new next time.

OK.

> This was a bug of the plugin. Fixed in v0.3.1. Please try it.

I tried benchmark_driver-output-gruff v0.3.1..

```
$  benchmark-driver test_gem.yaml --output gruff 
* empty...
  * red-datasets 0.0.3...
  * red-datasets 0.0.4...
* blank...
  * red-datasets 0.0.3...
  * red-datasets 0.0.4...
rendering graph...: graph.png
```

![graph_gem_ok](https://user-images.githubusercontent.com/706220/42513841-ebfe4e02-8492-11e8-845e-17fb08b5338a.png)

It worked fine.
Thank you for fixing!

> This is not a bug in both benchmark_driver and benchmark_driver-output-gruff.
> You just need install them in each Ruby versions.

Oh! Sorry! 
It is my mistake.

![graph_ruby_gems_ok](https://user-images.githubusercontent.com/706220/42513985-46a925e8-8493-11e8-82ab-a5d6329cf6d8.png)

It worked fine.
Wonderful!

Thank you!
